packages/trace-viewer/src/ui/uiModeView.tsx

```diff
@@ -93,6 +94,37 @@ export const UIModeView: React.FC<{}> = ({
   const [isDisconnected, setIsDisconnected] = React.useState(false);
   const [hasBrowsers, setHasBrowsers] = React.useState(true);
   const [testServerConnection, setTestServerConnection] = React.useState<TestServerConnection>();
+  const [settingsVisible, setSettingsVisible] = React.useState(false);
+  const [testingOptionsVisible, setTestingOptionsVisible] = React.useState(false);
+
+  const [runWorkers, setRunWorkers] = React.useState(queryParams.workers);
+  const singleWorkerSetting = React.useMemo(() => {
+    return [
+      runWorkers === '1',
+      (value: boolean) => {
+        // When started with `--workers=1`, the setting allows to undo that.
+        // Otherwise, fallback to the cli `--workers=X` argument.
+        setRunWorkers(value ? '1' : (queryParams.workers === '1' ? undefined : queryParams.workers));
```

Comment:
```suggestion
        setRunWorkers(value ? '1' : queryParams.workers);
```

Wouldn't this work as well? And then `runWorkers` would be the source of truth instead of always having also check `queryParams.workers`.